Lib/dataclasses.py

```diff
@@ -1549,11 +1550,29 @@ class C(Base):
         seen.add(name)
         annotations[name] = tp
 
+    def annotate_method(format):
+        typing = sys.modules.get("typing")
+        if typing is None and format == annotationlib.Format.FORWARDREF:
```

Comment:
We could also avoid importing typing for the SOURCE format here I think; is that worth it?

Reply:
I am not sure we can. I need `_convert_to_source`, there can be complex annotations that should be formatted properly. I will open a new issue about converting annotations to string with public API though. Right now I don't see a clear way.

Reply:
I opened https://github.com/python/cpython/issues/124412 for that.